src/styles/index.css

```diff
@@ -0,0 +1,114 @@
+/* stylelint-disable rule-empty-line-before */
+body {
+  margin: 0;
+}
+
+html {
+  height: 100%;
+  margin: 0;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+}
+
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  width: calc(7 * 100px + 6px);
+  padding: 10px;
+  gap: 1px;
+  counter-reset: day-counter 0;
+}
+
+.calendar__day {
+  width: 100px;
+  height: 100px;
+  justify-content: center;
+  align-items: center;
+  font-family: Arial, sans-serif;
+  font-size: 30px;
+  background-color: #eee;
+  border: 1px solid black;
+  display: flex;
+  box-sizing: border-box;
+}
+
+.calendar__day::before {
+  counter-increment: day-counter;
+  content: counter(day-counter);
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  background-color: #ffbfcb;
+  transform: translateY(-20px);
+  transition: all 0.5s;
+}
+
+.calendar--start-day-mon .calendar__day:nth-child(1) {
+  margin-left: calc(0 * (100px + 1px));
+}
+
+.calendar--start-day-tue .calendar__day:nth-child(1) {
+  margin-left: calc(1 * (100px + 1px));
+}
+
+.calendar--start-day-wed .calendar__day:nth-child(1) {
+  margin-left: calc(2 * (100px + 1px));
+}
+
+.calendar--start-day-thu .calendar__day:nth-child(1) {
+  margin-left: calc(3 * (100px + 1px));
+}
+
+.calendar--start-day-fri .calendar__day:nth-child(1) {
+  margin-left: calc(4 * (100px + 1px));
+}
+
+.calendar--start-day-sat .calendar__day:nth-child(1) {
+  margin-left: calc(5 * (100px + 1px));
+}
+
+.calendar--start-day-sun .calendar__day:nth-child(1) {
+  margin-left: calc(6 * (100px + 1px));
+}
```

Comment:
The styles for setting the margin-left of the first calendar day based on the start day of the week are duplicated. Consider consolidating these styles to improve maintainability. For example, you could use a single class with a variable or a calculation to determine the margin-left value.